internal/rbd/rbd_healer.go

```diff
@@ -0,0 +1,202 @@
+/*
+Copyright 2021 The Ceph-CSI Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rbd
+
+import (
+	"context"
+	"errors"
+	"sync"
+
+	"github.com/container-storage-interface/spec/lib/go/csi"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8s "k8s.io/client-go/kubernetes"
+
+	"github.com/ceph/ceph-csi/internal/util"
+)
+
+func accessModeStrToInt(mode string) csi.VolumeCapability_AccessMode_Mode {
+	switch mode {
+	case "ReadWriteOnce":
+		return csi.VolumeCapability_AccessMode_SINGLE_NODE_WRITER
+	case "ReadOnlyMany":
+		return csi.VolumeCapability_AccessMode_MULTI_NODE_READER_ONLY
+	case "ReadWriteMany":
+		return csi.VolumeCapability_AccessMode_MULTI_NODE_MULTI_WRITER
+	}
+	return csi.VolumeCapability_AccessMode_UNKNOWN
+}
+
+func getSecret(c *k8s.Clientset, ns, name string) (map[string]string, error) {
+	deviceSecret := make(map[string]string)
+
+	secret, err := c.CoreV1().Secrets(ns).Get(context.TODO(), name, metav1.GetOptions{})
+	if err != nil {
+		util.ErrorLogMsg("get secret failed, err: %v", err)
+		return deviceSecret, err
+	}
+	userID, ok := secret.Data["userID"]
```

Comment:
define userId,UserKey,encryptionPassphrase as variables and use it for fetching it from secret and also for logging and returning error messages. if something changes in the future we just need to change in one place.